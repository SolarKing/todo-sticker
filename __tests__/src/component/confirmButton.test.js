// @flow
import React from 'react';
import { shallow, type ShallowWrapper } from 'enzyme';
import ConfirmButton from '../../../src/component/confirmButton';
import { Modal } from 'antd';

describe('ConfirmButton', () => {
    let wrapper: ShallowWrapper;
    let handleOk = jest.fn();
    let config = {
        title: 'Are you sure?',
        okText: 'sure',
        cancelText: 'wait',
        onOk: handleOk,
    };
    beforeAll(() => {
        wrapper = shallow(
            <ConfirmButton
                config={config}
            />
        );
    });

    it('should render correct', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should execute confirm', () => {
        let confirm = jest.spyOn(Modal, 'confirm');
        wrapper.find('a').simulate('click');
        expect(confirm.mock.calls.length).toEqual(1);
        expect(confirm.mock.calls[0][0]).toEqual(config);
    });

    it('should execute onOk', () => {
        let confirm = jest.spyOn(Modal, 'confirm');
        wrapper.find('a').simulate('click');
        confirm.mock.calls[0][0].onOk();
        expect(handleOk.mock.calls.length).toEqual(1);
    });

});