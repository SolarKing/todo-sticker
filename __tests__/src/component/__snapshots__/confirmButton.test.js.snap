// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmButton should render correct 1`] = `
<a
  onClick={[Function]}
/>
`;
